### PR TITLE
feat: third party detail

### LINKED ISSUE #1631: Third Party Collection detail page
For this particular issue we need to:

- Create components that handle both the Decentraland and ThirdParty views for collections
- Add the UI for the buttons and context actions. Implement the ones we already have
- Add a new route to show the new component.

### PR DESCRIPTION
**Pull request by nicosantangelo**
image

Closes #1631

### CODE REVIEW

File: src/components/ThirdPartyCollectionDetailPage/ThirdPartyCollectionDetailPage.tsx [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+  }
+
+  renderPage() {
+    const { wallet, items } = this.props
+    const collection = this.props.collection!
+    const slots = 0
+
+    return (
+      <>
+        <Section className={collection.isPublished ? 'is-published' : ''}>
```
Comment (LautaroPetaccio) [inline src/components/ThirdPartyCollectionDetailPage/ThirdPartyCollectionDetailPage.tsx:53]: Third party collections shouldn't ever be published, so this class won't be used.

File: src/components/ThirdPartyCollectionDetailPage/ThirdPartyCollectionDetailPage.container.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+const mapState = (state: RootState): MapStateProps => {
+  const collectionId = getThirdPartyCollectionId(state) || ''
+  const collection = getCollection(state, collectionId)
+
+  return {
+    wallet: getWallet(state)!,
+    collection,
+    items: getCollectionItems(state, collectionId),
+    isOnSaleLoading: isOnSaleLoading(state),
```
Comment (LautaroPetaccio) [inline src/components/ThirdPartyCollectionDetailPage/ThirdPartyCollectionDetailPage.container.ts:23]: I don't find this variable to be in use, what is it for?

File: src/modules/translation/languages/zh.json [hunk 3]
```diff
@@ -410,6 +410,8 @@
     "blog_post": "博客",
     "cancel": "取消",
     "confirm": "确认",
+    "add": "添加",
+    "buy": "买",
```
Comment (LautaroPetaccio) [inline src/modules/translation/languages/zh.json:414]: `publish` seems to be missing here.

File: src/modules/translation/languages/en.json [hunk 4]
```diff
@@ -950,13 +951,19 @@
     "add_representation": "Add {bodyShape} representation",
     "change_item_file": "Edit representation"
   },
+  "third_party_collection_detail_page": {
+    "notice": "Each published item takes 1 Slot. {buy_link} to buy Slot Packs and start publishing your creations!",
+    "click_here": "Click here",
```
Comment (LautaroPetaccio) [inline src/modules/translation/languages/en.json:956]: This should be taken from the `global` translation.

File: src/modules/translation/languages/en.json [hunk 5]
```diff
@@ -950,13 +951,19 @@
...
+    "slots": "{amount, plural, =0 {No Slots} one {# Slot} other {# Slots}}",
+    "start_adding_items": "Looking good! Start adding items to your new collection",
+    "cant_remove": "You will not be able to add or remove items after publishing your collection."
```
Comment (LautaroPetaccio) [inline src/modules/translation/languages/en.json:959]: The user won't be able to remove already published items, but they will be able to remove unpublished items and add as many items as the slots let them.

File: src/components/ThirdPartyCollectionDetailPage/CollectionContextMenu/CollectionContextMenu.tsx [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+          </Button>
+        }
+        inline
+        direction="left"
+      >
+        <Dropdown.Menu>
+          <Dropdown.Item text={t('collection_context_menu.see_in_world')} onClick={this.handleNavigateToExplorer} />
+          <Dropdown.Item text={t('collection_context_menu.open_in_editor')} onClick={this.handleNavigateToEditor} />
+
+          {!isLocked(collection) ? (
```
Comment (LautaroPetaccio) [inline src/components/ThirdPartyCollectionDetailPage/CollectionContextMenu/CollectionContextMenu.tsx:53]: Can you add a TODO here to check for the items to see if any of them were already published? The idea is to disable the deletion of the collection if this happens.

File: src/modules/location/selectors.ts [hunk 7]
```diff
@@ -50,6 +50,18 @@ export const getCollectionId = (state: RootState) => {
...
+  {
+    collectionId: string
+  }
+>(locations.thirdPartyCollectionDetail())
+
+export const getThirdPartyCollectionId = (state: RootState) => {
+  const result = thirdPartyCollectionIdMatchSelector(state)
+  return result ? result.params.collectionId : null
+}
+
```
Comment (LautaroPetaccio) [inline src/modules/location/selectors.ts:64]: Can you add a test for this new selector?

**Review by LautaroPetaccio — CHANGES_REQUESTED** [review]

File: src/components/ThirdPartyCollectionDetailPage/ThirdPartyCollectionDetailPage.container.ts (hunk 2 above)
Comment (nicosantangelo) [reply src/components/ThirdPartyCollectionDetailPage/ThirdPartyCollectionDetailPage.container.ts:23]: Not used! it was for the on sale toggle, good eye

File: src/modules/translation/languages/en.json (hunk 5 above)
Comment (nicosantangelo) [reply src/modules/translation/languages/en.json:959]: True! leftover from the old translation

File: src/modules/location/selectors.ts (hunk 7 above)
Comment (nicosantangelo) [reply src/modules/location/selectors.ts:64]: Yes!

**Review by LautaroPetaccio — APPROVED** [review]

**Pull request merged**